analyzers/src/SonarAnalyzer.CSharp/Rules/DoNotUseOutRefParameters.cs

```diff
@@ -35,19 +35,18 @@ public sealed class DoNotUseOutRefParameters : SonarDiagnosticAnalyzer
         internal const string DiagnosticId = "S3874";
         private const string MessageFormat = "Consider refactoring this method in order to remove the need for this '{0}' modifier.";
 
-        private static readonly DiagnosticDescriptor rule =
-            DescriptorFactory.Create(DiagnosticId, MessageFormat);
+        private static readonly DiagnosticDescriptor Rule = DescriptorFactory.Create(DiagnosticId, MessageFormat);
 
-        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get; } = ImmutableArray.Create(rule);
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get; } = ImmutableArray.Create(Rule);
 
-        protected override void Initialize(SonarAnalysisContext context)
-        {
+        protected override void Initialize(SonarAnalysisContext context) =>
             context.RegisterSyntaxNodeActionInNonGenerated(
                 c =>
                 {
                     var parameter = (ParameterSyntax)c.Node;
 
-                    if (!parameter.Modifiers.Any(IsRefOrOut))
+                    if (!parameter.Modifiers.Any(IsRefOrOut)
+                        || (parameter.Parent.Parent as MethodDeclarationSyntax is { } method && IsDeconstructor(method)))
```

Comment:
The `Parent` is of `SyntaxNode` type and `is MethodDeclarationSyntax` returns true only if it's different than null and of type `MethodDeclarationSyntax`.

```suggestion
                        || (parameter.Parent.Parent is MethodDeclarationSyntax method && IsDeconstructor(method)))
```